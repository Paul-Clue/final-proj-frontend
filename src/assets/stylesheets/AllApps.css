@import url('https://fonts.googleapis.com/css2?family=Archivo+Black&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100&display=swap');

.appointments {
  position: absolute;
  top: 0vh;
  height: 90vh;
  left: 5vw;
  top: 5vh;
  width: 90vw;
  background-repeat: no-repeat;
  background-position-x: center;
  background-position-y: center;
  text-align: center;
}

.ul {
  list-style-type:none;
}

.backImage {
  position: absolute;
  left: -15vw;
  z-index: 0;
  width: 70vw;
}

.overlayDiv {
  background-color: rgba(86, 130, 3, .8);
  position: absolute;
  height: 90vh;
  left: 0vw;
  top: 0vh;
  width: 90vw;
}

.appList {
  font-family: Archivo Black, sans-serif;
  font-size: 1.5vw;;
  position: absolute;
  top: 25vh;
  left: 21vw;
  color: white;
}

.HOME {
  position: absolute;
  font-family: Poppins, sans-serif;
  top: 30vh;
  left: 5vw;
  color: white;
  font-size: 2vw;
  text-decoration: none;
  border: 1px solid white;
  background-color: rgba(86, 130, 3, .7);
}

.Apps {
  position: absolute;
  font-family:Archivo Black, sans-serif;
  font-size: 3vw;
  top: 16vh;
  left: 22vw;
  font-weight: bolder;
  color: white;
}

.AppsLine {
  position: absolute;
  top: 20vh;
  left: 11vw;
  width: 60vw;
  border-top: 2px solid white;
}

@media only screen and (min-width: 1500px) {
  .AppsLine {
    top: 22vh;
  }
}
