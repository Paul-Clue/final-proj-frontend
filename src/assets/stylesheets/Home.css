@import url('https://fonts.googleapis.com/css2?family=Bebas+Neue&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Archivo+Black&display=swap');

.header {
  position: absolute;
  font-family: Bebas Neue, sans-serif;
  font-weight: bolder;
  font-size: 11vw;
  top: 12vh;
  left: 30%;
}

.header2 {
  position: absolute;
  font-family: Archivo Black, sans-serif;
  font-weight: bolder;
  font-size: 3vw;
  top: 22vh;
  left: 25vw;
  color: gray;
}

.line {
  position: absolute;
  width: 32vw;
  height: 3px;
  top: 27vh;
  left: 37%;
  border-top: 2px dotted gray;
}

.accent {
  background-color: limegreen;
  width: 3vw;
  height: 8vh;
  position: absolute;
  right: 0;
  top: 50vh;
}

.accent2 {
  background-color: limegreen;
  width: 1vw;
  height: 8vh;
  position: absolute;
  left: 13vw;
  top: 50vh;
}

.card-group {
  width: 100vw;
}

.card-img-top {
  width: 10vw;
}

.card-body {
  width: 10vw;
}

.item {
  width: 15vw;
}

.home1 {
  width: 14vw;
}

.col {
  margin-top: 44vh;
  width: 14.2vw;
}

.card-text {
  font-size: 1vw;
}

.card-title {
  font-size: 2vw;
}

.icon {
  margin-top: 3vh;
  width: 1.5vw;
  opacity: 0.5;
}

.burb {
  margin-bottom: -2vh;
  background: rgba(40, 250, 330, 0.7);
}

.dg {
  margin-bottom: -2vh;
  background: rgba(200, 150, 310, 0.7);
}

.arm {
  margin-bottom: -3vh;
}

.ver {
  margin-bottom: -4vh;
}

.guc {
  margin-bottom: -2vh;
  background: rgba(100, 130, 310, 0.9);
}

.tom {
  margin-bottom: -2vh;
}

@media only screen and (min-width: 1100px) {
  .header2 {
    top: 24vh;
  }

  .line {
    top: 29vh;
  }
}

@media only screen and (min-width: 1200px) {
  .header2 {
    top: 26vh;
  }

  .line {
    top: 31vh;
  }
}

@media only screen and (min-width: 1500px) {
  .header2 {
    top: 35vh;
  }

  .line {
    top: 40vh;
  }
}

@media only screen and (min-width: 1800px) {
  .header2 {
    top: 34vh;
  }

  .line {
    top: 40vh;
  }
}
