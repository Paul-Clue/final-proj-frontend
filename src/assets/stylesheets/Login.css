@import url('https://fonts.googleapis.com/css2?family=Bebas+Neue&family=Lobster&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Archivo+Black&display=swap');

.loginDiv {
  width: 100vw;
  height: 100vh;
  border: 2px solid black;
}

.loginDiv2 {
  position: absolute;
  top: 5vh;
  left: 5vw;
  width: 90vw;
  height: 90vh;
  background-color: orange;
  background-repeat: no-repeat;
  background-position-x: center;
  background-position-y: center;
}

.bigLogo {
  font-family: Lobster, serif;
  color: white;
  font-size: 7vw;
  margin-top: 25vh;
  margin-left: 30vw;
}

form {
  color: white;
  font-family: Archivo white, sans-serif;
  font-weight: bolder;
  font-size: 1.5vw;
  margin-left: 19vw;
}

.submitB {
  background-color: rgba(86, 130, 3, 0.7);
  color: white;
  border-radius: 40%;
  border: none;
  height: 6vh;
}

.err {
  color: red;
  font-size: 3vw;
}
