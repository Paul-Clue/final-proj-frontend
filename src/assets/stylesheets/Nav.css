@import url('https://fonts.googleapis.com/css2?family=Bebas+Neue&family=Lobster&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Archivo+Black&display=swap');

.NavHead {
  font-family: Lobster, serif;
  font-size: 2vw;
}

.nav {
  background-color: whitesmoke;
  width: 14vw;
  height: 100vh;
  border-right: 3px solid lightgray;
}

.home {
  margin-top: 30vh;
  text-align: center;
  width: 7vw;
  height: 3vw;
}

.home:hover {
  background-color: limegreen;
  text-align: center;
  border-radius: 5%;
}

.Link {
  text-decoration: none;
  color: black;
  opacity: 78%;
  font-family: Archivo Black, sans-serif;
  font-weight: bolder;
  font-size: 1.5vw;
}

.appoint {
  margin-top: 5vh;
  margin-left: -2vw;
  text-align: center;
  width: 12vw;
  height: 3vw;
}

.appoint:hover {
  background-color: limegreen;
  text-align: center;
  border-radius: 5%;
}

.icon2 {
  width: 2vw;
}

.iconDiv {
  margin-top: 38vh;
}

@media only screen and (min-width: 1000px) {
  .appoint {
    margin-left: 0;
  }
}
