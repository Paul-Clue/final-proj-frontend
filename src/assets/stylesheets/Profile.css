@import url('https://fonts.googleapis.com/css2?family=Archivo+Black&family=M+PLUS+1p:wght@300&display=swap');

.logo1 {
  margin-right: 2vw;
  width: 12vw;
}

.makeAppoint {
  font-size: 1vw;
}

.words {
  position: absolute;
  right: 1vw;
  top: 5vh;
  text-align: center;
  font-family: M PLUS, sans-serif, serif;
  font-weight: bolder;
}

.imgDiv {
  margin-top: 10vh;
}

.frame {
  width: 45vw;
  position: absolute;
  top: 30vh;
  left: 28vw;
}

.submit {
  background: limegreen;
  border: none;
  border-radius: 45%;
  height: 3vw;
  width: 15vw;
}

.design {
  text-align: center;
}

.accent1 {
  width: 3vw;
  height: 8vh;
  background-color: limegreen;
  position: absolute;
  left: 14vw;
  top: 69vh;
}

.colors {
  width: 13vw;
  margin-top: -7vh;
}
