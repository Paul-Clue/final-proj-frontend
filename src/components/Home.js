// import React, { useState, useEffect } from 'react';
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { getData } from '../util/apiFetch';
import { addFrame } from '../actions';
import Nav from './Nav';

function Home(props) {
  const { routerProps } = props;
  const dispatch = useDispatch();

  const url = 'http://localhost:3001/frm';
  useEffect(() => {
    getData(url)
      .then((response) => response.json())
      .then((data) => {
        // setCurrentUser(data);
        console.log(data)//eslint-disable-line
        dispatch(addFrame(data));

        routerProps.history.push('/');
        return data;
      })
      .then((data) => console.log(data));//eslint-disable-line
  }, []);

  // const { currentuser } = props;
  const frame2 = useSelector((state) => state.frame);
  console.log(frame2)//eslint-disable-line

  return (
    <>
      <Nav />
      <ul>
        {frame2.map((frm) => (
          <li key={frm.id}>
            {frm.make}
            <img src={frm.pic} alt="frames" />
          </li>
        ))}
      </ul>
    </>
  );
}

Home.propTypes = { routerProps: PropTypes.instanceOf(Object).isRequired };

export default Home;
