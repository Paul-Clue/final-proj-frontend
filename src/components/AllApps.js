// import React, { useState, useRef, useEffect } from 'react';
import React, { useEffect } from 'react';
// import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
// import qs from 'qs';
// import { createBrowserHistory } from 'history';
// import { postData, getData2 } from '../util/apiFetch';
import { getData2 } from '../util/apiFetch';
// import { postData } from '../util/apiFetch';
import { setDate } from '../actions';
import Nav from './Nav';
import '../assets/stylesheets/Profile.css';

function AllApps() {
  const currentuser = useSelector((state) => state.user);
  // const dat = useRef();

  // console.log(props)//eslint-disable-line

  const date2 = useSelector((state) => state.appointments);
  // console.log(date2)//eslint-disable-line
  // const [date, setDates] = useState(date2.date);

  const frm = useSelector((state) => state.prof);
  // const [frm2] = useState(frm);
  // console.log(frm2)//eslint-disable-line
  // console.log(frm)//eslint-disable-line
  const dispatch = useDispatch();
  // const { routerProps } = props;

  const param = {
    user_id: currentuser.id,
    frame_id: frm.id,
  };

  const url = 'http://localhost:3001/appoints2';
  useEffect(() => {
    getData2(url, param)
      .then((response) => {
        if (response.ok) {
          // console.log(response)//eslint-disable-line
          // console.log("That")//eslint-disable-line
          return response.json();
        }
        console.log(response)//eslint-disable-line
        console.log("That2")//eslint-disable-line
        return '';
      })
      .then((data) => {
        // console.log(data)//eslint-disable-line
        console.log(data)//eslint-disable-line
          console.log("List of things:")//eslint-disable-line
        dispatch(setDate(data));

        // routerProps.history.push('/profile');
        return data;
      })
      .then((data) => console.log(data) + console.log(date2));
  }, []);
  // const show = [];
  // for (const [key, value] of Object.entries(date2)) {//eslint-disable-line
  //   // <h1 key={key}>{value}</h1>
  //   show.push([value, value.city, value.date]);
  // }

  return (
    <>
      <Nav />
      {/* {show.filter((date) => <h1 key={date[0]}>{date[0]}</h1>)} */}

      {/* <h1>{date2[0].id}</h1> */}
      {console.log('This is above')}
      {/* {console.log(show)} */}
      {console.log(date2)}
      {/* {console.log(Array.isArray(show))} */}
      {console.log('This is under')}
    </>
  );
}

// AllApps.propTypes = { currentuser: PropTypes.instanceOf(Object).isRequired };
// AllApps.propTypes = { routerProps: PropTypes.instanceOf(Object).isRequired };

export default AllApps;
