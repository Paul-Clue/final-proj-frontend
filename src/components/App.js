// import React, { useState } from 'react';
import React from 'react';
// import { useSelector, useDispatch } from 'react-redux';
import { useSelector } from 'react-redux';
import {
  BrowserRouter,
  withRouter,
  Route,
  Switch,
} from 'react-router-dom';
import '../App.css';
import Login from '../conainers/Login';
import Profile from './Profile';
import Home from './Home';
// import { setUser } from '../actions';

function App() {
  // constructor() {
  //   super();
  //   this.state = {
  //     CurrentUser: null,
  //   };
  // }
  // const [CurrentUser, setCurrentUser] = useState({ name: null, password: null });
  const CurrentUser = useSelector((state) => state.user);
  console.log(`This is current user: ${CurrentUser}`)//eslint-disable-line
  console.log(CurrentUser)//eslint-disable-line
  // const use = useSelector((state) => state.user);
  // const dispatch = useDispatch();

  // const userSet = (data) => {
  //   // setCurrentUser({
  //   //   name: data.name,
  //   //   password: data.password,
  //   // });
  //   dispatch(setUser(data));
  // };

  // const { CurrentUser } = this.state;
  return (
    <>
      <BrowserRouter>
        <Switch>
          <Route exact path="/" component={Home} />

          <Route exact path="/login" render={(props) => <Login routerProps={props} />} />
          <Route
            exact
            path="/Profile"
            render={(props) => (
              CurrentUser
                ? (<Profile currentuser={CurrentUser} routerProps={props} />)
                : (<Login />)
            )}
          />
        </Switch>
      </BrowserRouter>
    </>
  );
}

export default withRouter(App);
// export default App;
