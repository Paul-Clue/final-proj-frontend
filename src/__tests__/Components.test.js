import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from '@reduxjs/toolkit';
import configureStore from 'redux-mock-store';
import ErrReducer from '../reducers/error';
import FrameReducer from '../reducers/frame';
import ProfileReducer from '../reducers/profile';
import UserReducer from '../reducers/user.js'
import DateReducer from '../reducers/date.js';
import rootReducer from '../reducers/index';
import App from '../components/App';
import AllApps from '../components/AllApps';
import Home from '../components/Home';
import Login from '../conainers/Login';
import Profile from '../components/Profile';
import Nav from '../components/Nav';

// const coins = [{
//   name: 'bitcoin',
// }];

// const filters = 'All';

// const details = [{
//   name: 'testing',
// }];

// const mockStore = configureStore({
//   reducer: { rootReducer }, addCoin: coins, filter: filters, detail: details,
// });

const user = {
  name: '',
  password: '',
  user_id: '',
};

const frame = [];

const appointments = [];

const prof = {
  make: '',
  pic: '',
};

const error = '';

const initialState = {
  frame,
  user,
  prof,
  error,
  appointments,
};

const mockStore = configureStore([]);

// const mockStore = configureStore({
//   reducer: {rootReducer}, user, appointments, frame, prof, error,
// });

describe('AllApps Component', () => {
  let store;
  beforeEach(() => {
    // store = createTestStore();
    store = mockStore(initialState);
    store.dispatch = jest.fn();
  });
  test('It should render', () => {
    const all = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <AllApps />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(all).toMatchSnapshot();
  });
});

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    console.log(error, errorInfo);//eslint-disable-line
  }

  render() {
    if (this.state.hasError) {//eslint-disable-line
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children;//eslint-disable-line
  }
}

export function createTestStore() {//eslint-disable-line
  const store = createStore(
    combineReducers({
      user: UserReducer,
      appointments: DateReducer,
      frame: FrameReducer,
      prof: ProfileReducer,
      error: ErrReducer,
    }),
  );
  return store;
}

let store;

describe('App Component', () => {
  test('It should render', () => {
    const app = renderer.create(
      <BrowserRouter>
        <ErrorBoundary>
          <Provider store={store}>
            <App />
          </Provider>
        </ErrorBoundary>
      </BrowserRouter>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});

describe('Home Component', () => {
  test('It should render', () => {
    const home = renderer.create(
      <BrowserRouter>
        <ErrorBoundary>
          <Provider store={store}>
            <Home />
          </Provider>
        </ErrorBoundary>
      </BrowserRouter>,
    ).toJSON();
    expect(home).toMatchSnapshot();
  });
});

describe('Login Component', () => {
  test('It should render', () => {
    const login = renderer.create(
      <BrowserRouter>
        <ErrorBoundary>
          <Login />
        </ErrorBoundary>
      </BrowserRouter>,
    ).toJSON();
    expect(login).toMatchSnapshot();
  });
});

describe('Nav Component', () => {
  beforeEach(() => {
    store = createTestStore();
  });
  test('It should render', () => {
    const nav = renderer.create(
      <Provider store={store}>
        <Nav />
      </Provider>,
    ).toJSON();
    expect(nav).toMatchSnapshot();
  });
});

// describe('Profile Component', () => {
//   beforeEach(() => {
//     store = mockStore(initialState);

//     store.dispatch = jest.fn();

//     component = renderer.create(
//       <Provider store={store}>
//         <ErrorBoundary>
//           <Profile />
//         </ErrorBoundary>
//       </Provider>,
//   );
// });

//   it('Should render', () => {
//     expect(component.toJSON()).toMatchSnapshot();
//   });
// }); 

describe('Profile Component', () => {
  beforeEach(() => {
    store = createTestStore();
  });
  test('It should render', () => {
    const prof = renderer.create(
      <Provider store={store}>
        <ErrorBoundary>
          <Profile />
        </ErrorBoundary>
      </Provider>,
    ).toJSON();
    expect(prof).toMatchSnapshot();
  });
});
