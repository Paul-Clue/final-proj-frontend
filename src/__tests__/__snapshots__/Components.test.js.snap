// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllApps Component It should render 1`] = `
<div
  className="appointments"
  style={
    Object {
      "backgroundImage": "url( tomford.jpg)",
    }
  }
>
  <div
    className="overlayDiv"
  >
    <h1
      className="Apps"
    >
      Your Appointment Dates
    </h1>
    <hr
      className=" AppsLine"
    />
    <a
      className="HOME"
      href="/Home"
      onClick={[Function]}
    >
      <button
        className="HOME"
        type="button"
      >
        HOME
      </button>
    </a>
    <ul
      className="ul"
    />
  </div>
</div>
`;

exports[`App Component It should render 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`Home Component It should render 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`Login Component It should render 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`Nav Component It should render 1`] = `
<nav
  className="nav flex-column navbar-light "
>
  <div
    className="container-fluid"
  >
    <span
      className="navbar-brand mb-0 h1"
    >
      <h1
        className="NavHead"
      >
        FrameFace
      </h1>
    </span>
    <div
      className="home"
    >
      <a
        className="Link home"
        href="/Home"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
    <div
      className="appoint"
    >
      <a
        className="Link login1"
        href="/Appoint"
        onClick={[Function]}
      >
        Appointments
      </a>
    </div>
    <div
      className="iconDiv"
    >
      <img
        alt="icon"
        className="icon2"
        src="twitter3.png"
      />
      <img
        alt="icon"
        className="icon2"
        src="facebook2.png"
      />
      <img
        alt="icon"
        className="icon2"
        src="pinterest.png"
      />
      <img
        alt="icon"
        className="icon2"
        src="google.png"
      />
    </div>
  </div>
</nav>
`;

exports[`Profile Component It should render 1`] = `
<h1>
  Something went wrong.
</h1>
`;
